src/apps/beefy/shortcuts.ts

```diff
@@ -0,0 +1,121 @@
+import { Address, parseUnits, erc20Abi, encodeFunctionData } from 'viem'
+import { z } from 'zod'
+import { logger } from '../../log'
+import { getClient } from '../../runtime/client'
+import {
+  simulateTransactions,
+  UnsupportedSimulateRequest,
+} from '../../runtime/simulateTransactions'
+import { ZodAddressLowerCased } from '../../types/address'
+import { NetworkId } from '../../types/networkId'
+import {
+  ShortcutsHook,
+  createShortcut,
+  tokenAmounts,
+  Transaction,
+} from '../../types/shortcuts'
+import { poolAbi } from './abis/pool'
+
+// Hardcoded fallback if simulation isn't enabled
+const DEFAULT_DEPOSIT_GAS = 500_000n
+// Padding we add to simulation gas to ensure we specify enough
+const SIMULATED_DEPOSIT_GAS_PADDING = 250_000n
```

Comment:
is this required for beefy?

Reply:
looks like it is estimating around 640,000 and then actually uses somewhere between 600,000 and 650,000. Set the padding to 100,000 but can lower or remove it if that is unnecessary, not sure how much padding we wanted.